### PR TITLE
Backend/feat/#1361/cleanup stuck rides bookings flow

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
To be merged with https://github.com/nammayatri/shared-kernel/pull/157

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [x] This PR contains API breaking changes - removed unused multipleRideEnd and mutipleRideCancel apis

## How did you test it?
tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs [hunk 1]
```diff
@@ -364,6 +364,7 @@ onUpdate ValidatedBookingCancelledReq {..} = do
   fork "incrementing fraud counters" $ do
     mFraudDetected <- SMC.anyFraudDetected booking.riderId booking.merchantId merchantConfigs
     whenJust mFraudDetected $ \mc -> SMC.blockCustomer booking.riderId (Just mc.id)
+  -- TODO  If Ride doesn't exist in BAP, create a cancelled ride entry.
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs:367]: @roman-bodavskiy Did you miss this todo by mistake ?
OR you will do the changes later ?

Comment (roman-bodavskiy) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs:367]: @RitikaHanish this TODO require changes in on_update api. Working on this now. Can be merged in separate PR

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs [hunk 2]
```diff
@@ -15,31 +15,38 @@
...
+-- we can receive more rideInfo in later PRs
+data RideInfo = RideInfo
+  { bppRideId :: Id DRide.BPPRide,
     rideStatus :: DRide.RideStatus
   }
 
-onStatus :: EsqDBFlow m r => OnStatusReq -> m ()
+onStatus :: (EsqDBFlow m r, Esq.EsqDBReplicaFlow m r) => OnStatusReq -> m ()
 onStatus OnStatusReq {..} = do
   booking <- QBooking.findByBPPBookingId bppBookingId >>= fromMaybeM (BookingDoesNotExist $ "BppBookingId: " <> bppBookingId.getId)
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs:48]: We can run this in replica

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs [hunk 3]
```diff
@@ -15,31 +15,38 @@
...
-  ride <- QRide.findByBPPRideId bppRideId >>= fromMaybeM (RideDoesNotExist $ "BppRideId: " <> bppRideId.getId)
+  ride <- QRide.findByBPPRideId rideInfo.bppRideId >>= fromMaybeM (RideDoesNotExist $ "BppRideId: " <> rideInfo.bppRideId.getId)
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnStatus.hs:49]: run In replica

File: Backend/app/dashboard/CommonAPIs/src/Dashboard/Common.hs [hunk 4]
```diff
@@ -67,3 +69,32 @@ data Summary = Summary
...
+listItemTaggedObject :: SumEncoding
+listItemTaggedObject =
+  TaggedObject
+    { tagFieldName = "result",
+      contentsFieldName = "errorMessage"
+    }
+
+-- is it correct to show every error?
+listItemErrHandler :: Monad m => SomeException -> m ListItemResult
+listItemErrHandler = pure . FailItem . show @Text @SomeException
```
Comment (roman-bodavskiy) [inline Backend/app/dashboard/CommonAPIs/src/Dashboard/Common.hs:100]: is it correct to show every error?

**Review by hkmangla — APPROVED** [review]

**Pull request merged**